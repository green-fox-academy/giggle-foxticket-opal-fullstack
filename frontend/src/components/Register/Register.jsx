import React from 'react';
import useCustomForm from '../../hooks/useCustomForm';
import './Register.styles.sass';
import Button from '../Button/Button';
import axios from 'axios';

const initialValues = {
  email: '',
  username: '',
  password: '',
  confirm: '',
};

const Register = () => {
  const { values, handleChange, handleSubmit } = useCustomForm({
    initialValues,
    onSubmit: values => {
      values.values.password === values.values.confirm
        ? axios({
            method: 'post',
            baseURL: 'http://localhost:3000/',
            url: '/api/users',
            data: {
              name: values.values.username,
              email: values.values.email,
              password: values.values.password,
            },
          })
            .then(res => console.log(res))
            .catch(err => console.log(err))
        : console.log(`Passwords don't match!`);
    },
  });
  return (
    <div className="form-container">
      <form onSubmit={handleSubmit} className="register-form">
        <fieldset>
          <h1>Foxticket</h1>
          <input
            type="email"
            name="email"
            onChange={handleChange}
            value={values.email}
            placeholder="E-mail..."
            autoComplete="off"
            required={true}
          />

          <input
            type="text"
            name="username"
            onChange={handleChange}
            value={values.username}
            placeholder="User Name..."
            autoComplete="off"
            required={true}
          />

          <input
            type="password"
            name="password"
            onChange={handleChange}
            value={values.password}
            placeholder="Password..."
            autoComplete="off"
            required={true}
          />

          <input
            type="password"
            name="confirm"
            onChange={handleChange}
            value={values.confirm}
            placeholder="Confirm password..."
            autoComplete="off"
            required={true}
          />

          <Button type="submit" buttonSize="btn--medium">
            Sign Up
          </Button>
        </fieldset>
      </form>
    </div>
  );
};

export default Register;
