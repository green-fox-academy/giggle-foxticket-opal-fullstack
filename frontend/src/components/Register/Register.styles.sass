.form-container
  display: flex
  align-items: center
  justify-content: center
  height: 100vh

  .register-form
    width: 600px
    background: #282828
    border-radius: 10px
    border: 0.1px solid red

  h1
    font-size: 55px
    margin-top: 30.2px
    width: fit-content
    height: fit-content

  fieldset
    padding: 10px 20px 20px 20px
    display: flex
    justify-content: center
    border: none

  input
    margin: 10px 10px 10px 0
    padding: 15px
    font-size: 16px
    box-sizing: border-box
    width: 100%
    display: flex
    flex-wrap: wrap

  button
    margin-left: 452px
