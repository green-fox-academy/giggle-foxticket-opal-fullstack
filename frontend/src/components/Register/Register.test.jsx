import React from 'react';
import Register from './Register';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../flux/store';

describe('Register Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Register />
      </Provider>
    );
    expect(wrapper.contains(<Register />)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
