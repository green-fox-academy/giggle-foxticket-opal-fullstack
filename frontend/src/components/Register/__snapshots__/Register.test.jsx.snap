// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component should match the snapshot 1`] = `
<div
  className="register-body"
>
  <div
    className="register_title"
  >
    Foxticket
  </div>
  <div
    className="register_container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <input
        id="confirm"
        name="confirm"
        onChange={[Function]}
        placeholder="Confirm password"
        required={true}
        type="password"
        value=""
      />
      <button
        className="register_btn"
      >
        Register
      </button>
    </form>
  </div>
</div>
`;
